lib/api/kuzzle.js

```diff
@@ -0,0 +1,159 @@
+  RemoteActionsController = require('./controllers/remoteActionsController'),
+  Repositories = require('./core/models/repositories'),
+  RouterController = require('./controllers/routerController'),
+  Services = require('../services'),
+  Statistics = require('./core/statistics'),
+  TokenManager = require('./core/auth/tokenManager');
+
+/** @typedef {{
+ *   request: {
+ *     maxRetainedRequests: {Number},
+ *     maxConcurrentRequests: {Number},
+ *     roleWithoutAdmin: *
+ *   },
+ *   server: Boolean,
+ *   httpPort: Number,
+ *   internalIndex: String,
+ *   pluginsManager: {
+ *     pipeWarnTime: Number,
+ *     pipeTimeout: Number,
+ *     dataCollection: String
+ *   },
+ *   userProfiles: {},
+ *   userRoles: {},
+ *   jsonWebToken: {
+ *     secret: String
+ *   },
+ *   roleWithoutAdmin: {
+ *     _id: String
+ *   }
+ * }} Params
+ */
+
+/**
+ * @this {KuzzleServer}
+ * @constructor
+ */
+function KuzzleServer () {
+  /** @type {Params} */
+  this.rawParams = rc('kuzzle');
+
+  this.config = config(this.rawParams);
+  this.config.apiVersion = packageInfo.apiVersion;
+  this.config.version = packageInfo.version;
+
+  this.hooks = new Hooks(this);
+  this.services = new Services(this);
+  this.remoteActions = new RemoteActions(this);
+  this.internalEngine = new InternalEngine(this);
+  this.pluginsManager = new PluginsManager(this);
+  this.tokenManager = new TokenManager(this);
+  this.indexCache = new IndexCache(this);
+  this.repositories = new Repositories(this);
+
+  this.passport = new PassportWrapper();
+
+
+  // The funnel controller dispatch messages between the router controller and other controllers
+  this.funnel = new FunnelController(this);
+
+  // The router controller listens to client requests and pass them to the funnel controller
+  this.router = new RouterController(this);
+
+  // Room subscriptions core components
+  this.hotelClerk = new HotelClerk(this);
+  this.dsl = new Dsl(this);
+
+  // Notifications core component
+  this.notifier = new Notifier(this);
+
+  // Statistics core component
+  this.statistics = new Statistics(this);
+
+  // Create and init the Cluster
+  this.entryPoints = new EntryPoints(this, this.rawParams);
+
+  // The remote actions controller listens to remotes actions from other processes (like the CLI) and passes them through the internal broker
+  this.remoteActionsController = new RemoteActionsController(this);
+
+  /**
+   * Initializes all the needed components of a Kuzzle Server instance.
+   *
+   * By default, this script runs a standalone Kuzzle Server instance:
+   *   - Internal services
+   *   - Controllers
+   *   - Hooks emitters
+   *
+   * @this {KuzzleServer}
+   * @param {Object} params command line and/or configuration file arguments
+   *                        overrides the 'feature' argument
+   * @param {Object} [feature={}] allow to programatically tune what part of Kuzzle you want to run
+   */
+  this.start = function (params, feature) {
```

Comment:
NB: I've just updated PR #356 with some clean-ups in `KuzzleServer.start()` method (see https://github.com/kuzzleio/kuzzle/pull/356/commits/fab5b90e46974c3bc6af6d4df4ae459bd01a863e )
These changes should be reported here also ;)